tests/test_app.py

```diff
@@ -8,6 +9,20 @@ def test_init():
     assert app
 
 
+def test_run():
```

Comment:
Oh, now I see you are actually testing 2 things here. The return value and the fact that this was run. Please split it to 2 tests. One test should assure that `run_internally` was being called. The other one should  care for the exit code and can use a lambda statement instead of mock...